azext_iot/central/services/device.py

```diff
@@ -84,6 +85,47 @@ def list_devices(
     return devices
 
 
+def get_device_registration_summary(
+    cmd, app_id: str, token: str, central_dns_suffix="azureiotcentral.com",
+):
+    """
+    Get device registration summary for a given app
+
+    Args:
+        cmd: command passed into az
+        app_id: name of app (used for forming request URL)
+        token: (OPTIONAL) authorization token to fetch device details from IoTC.
+            MUST INCLUDE type (e.g. 'SharedAccessToken ...', 'Bearer ...')
+        central_dns_suffix: {centralDnsSuffixInPath} as found in docs
+
+    Returns:
+        registration summary
+    """
+
+    registration_summary = {
+        DeviceStatus.provisioned.value: 0,
+        DeviceStatus.registered.value: 0,
+        DeviceStatus.unassociated.value: 0,
+        DeviceStatus.blocked.value: 0,
+    }
+
+    url = "https://{}.{}/{}".format(app_id, central_dns_suffix, BASE_PATH)
+    headers = _utility.get_headers(token, cmd)
+
+    while url is not None:
+        response = requests.get(url, headers=headers)
```

Comment:
reminder: as we discussed offline maybe worth a `logger.info("Processed {} devices...".format(sum(registration.summary.keys()))`